### DISCUSSION TITLE
How do we handle ExtraTables?

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [SuperOffice/typescript-for-superoffice#25] [General]
I was expecting to find something like: soApi.getExtraTableAgent() or similar.
How can we create/load/update extratables?

Regards
Pär

**Comment by ejfasting** [comment]
Hello @Nawksen, 

In the REST API you would use the [DatabaseTableAgent](https://docs.superoffice.com/en/api/netserver/web-services/howto/custom-objects/rest-add-custom-object-row.html?tabs=DatabaseTableAgent)to read/insert/update rows from an extraTable. 

Example:
```typescript
var temp = soApi.getDatabaseTableAgent();
var result = await temp.readRowAsync("y_dummy", 1); //First parameter is table name, the second is id to read. Similar to what you pass inn to the [REST API](https://docs.superoffice.com/en/api/netserver/web-services/howto/custom-objects/rest-get-custom-object-row.html?tabs=DatabaseTableAgent)

const obj = JSON.stringify(result, null, 2);
context.result.body = obj;
```

**Reply by PPetterssonSO** [reply]
Works perfect, but, may I please have an example on how to create a new row and populate a field as well please?

**Reply by ejfasting** [reply]
Sure:

```typescript
var databaseTableAgent = soApi.getDatabaseTableAgent();

//This is the tableRow-object you need to pass inn. Note that it is (almost) identical to the [sample on docs](https://docs.superoffice.com/en/api/netserver/web-services/howto/custom-objects/rest-add-custom-object-row.html?tabs=DatabaseTableAgent). 
var newTableRow = 
{
    "tableName": "y_dummy",
    "values": {
        "x_text": "From typescript"
        } 
};

var result = await databaseTableAgent.insertRowAsync(newTableRow); 
const obj = JSON.stringify(result, null, 2);
context.result.body = obj;

```

You then get the id of the new row back (in my case Id = 4):
![image](https://github.com/user-attachments/assets/8480a30a-8f44-4df3-a329-9345156b8c7c)


//Eivind

**Reply by PPetterssonSO** [reply]
You are now officially my favorite Norwegian!

**Reply by ejfasting** [reply]
Glad to hear it! :) 

FYI I've added samples for read/insert/delete/update to [the repo](https://github.com/SuperOffice/typescript-for-superoffice/tree/main/Examples/Entities/CustomObjects)

Have a SUPER day!

//Eivind
